src/rules/completedDocsRule.ts

```diff
@@ -19,100 +19,415 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+export interface IBlockRequirementDescriptor {
+    visibilities?: Visibility[];
+}
+
+export interface IClassRequirementDescriptor {
+    locations?: Location[];
+    privacies?: Privacy[];
+}
+
+export type RequirementDescriptor = IBlockRequirementDescriptor | IClassRequirementDescriptor;
+
+export interface IRequirementDescriptors {
+    [type: string /* DocType */]: RequirementDescriptor;
+}
+
+export type All = typeof Rule.ALL;
+
+export type DocType = All
+    | typeof Rule.ARGUMENT_CLASSES
```

Comment:
Did you meant to use `const` instead of `public static`? In that case, I see the string literal type.